### PR TITLE
bump sbt-protoc & fix compatibility with other sbt-protoc plugins

### PR DESCRIPTION
**Pull request by bjaglin**
This is a second take at https://github.com/akka/akka-grpc/pull/696, which had silently regressed because of https://github.com/akka/akka-grpc/pull/702.

As described in the commit messages:
* cc1c166 fixes and enforce the compatibility with other plugins affecting sbt-protoc keys in `Compile` (what should have been done in https://github.com/akka/akka-grpc/pull/696)
* 415866b bumps sbt-protoc (which does not need any change)
* 66850c0 makes use of https://github.com/thesamet/sbt-protoc/blob/master/CHANGELOG.md#09931 to extend the support of cc1c166 to `Test`
* 2bd5204 is a bit unrelated, but makes more explicit what is needed for `IntegrationTest` to work as expected (as already tested by https://github.com/akka/akka-grpc/blob/v0.8.2/sbt-plugin/src/sbt-test/gen-scala-server/03-test-config/build.sbt#L10-L14)

I chose to group all these commits in the same PR as the sbt-protoc bump is mostly useful for configuration improvements.

### CODE REVIEW

File: docs/src/main/paradox/buildtools/sbt.md [hunk 1]
```diff
@@ -25,7 +25,9 @@ you have to inherit the `.proto` definitions from `Compile` over to `Test`:
 If you have other configurations with `.proto` sources (for example `IntegrationTest`), you can enable the plugin on them:
 
 ```
-.settings(AkkaGrpcPlugin.configSettings(IntegrationTest))
+configs(IntegrationTest)
+Defaults.itSettings
+AkkaGrpcPlugin.configSettings(IntegrationTest))
```
Comment (raboof) [inline docs/src/main/paradox/buildtools/sbt.md:30]: Good improvement. Perhaps this should be a snippet from a scripted test?

**Comment by raboof** [comment]
OK TO TEST

File: docs/src/main/paradox/buildtools/sbt.md (hunk 1 above)
Comment (bjaglin) [reply docs/src/main/paradox/buildtools/sbt.md:30]: Indeed, I was not familiar with the syntax but it looks easy - I'll amend.

Comment (bjaglin) [reply docs/src/main/paradox/buildtools/sbt.md]: cf92528

Comment (bjaglin) [reply docs/src/main/paradox/buildtools/sbt.md]: 2bd5204

**Pull request merged**